src/request.js

```diff
@@ -1,25 +1,28 @@
 import mitt from 'mitt';
-import {
-  convertBufferToLines
-} from './utils';
+import { convertBufferToLines } from './utils';
 
-const fetcher = Promise
-  .resolve()
-  .then(() => 'fetch' in self ? self.fetch : import('whatwg-fetch').then(() => self.fetch));
+const fetcher = Promise.resolve().then(
+  () =>
+    'fetch' in self ? self.fetch : import('whatwg-fetch').then(() => self.fetch)
+);
 
-export const request = (url, options) => {
+export default (url, options) => {
   const emitter = mitt();
 
   emitter.on('start', async () => {
     try {
       const fetch = await fetcher;
-      const response = await fetch(url, Object.assign({ credentials: 'omit' }, options));
+      const response = await fetch(
```

Comment:
I'm not sure `fetch` throws an error if something goes wrong. It will only set the `response.ok` to false. The try catch statement might not be needed after all.

Reply:
It will for network errors.